xmtp_mls/src/identity_updates.rs

```diff
@@ -148,8 +152,26 @@ where
             .get_association_state(conn, inbox_id.as_ref(), starting_sequence_id)
             .await?;
 
-        let incremental_updates = conn
-            .get_identity_updates(inbox_id, starting_sequence_id, ending_sequence_id)?
+        let incremental_updates = conn.get_identity_updates(
+            &inbox_id.as_ref(),
+            starting_sequence_id,
+            ending_sequence_id,
+        )?;
+
+        let last_sequence_id = incremental_updates.last().map(|update| update.sequence_id);
+        if ending_sequence_id.is_some()
+            && last_sequence_id.is_some()
+            && last_sequence_id != ending_sequence_id
+        {
+            log::error!(
+                "Did not find the expected last sequence id. Expected: {:?}, Found: {:?}",
+                ending_sequence_id,
+                last_sequence_id
+            );
+            return Err(AssociationError::MissingIdentityUpdate.into());
```

Comment:
This is a new error scenario I think is possible in theory, maybe more so once replication is implemented. When we get a group metadata update for a specific sequence ID, it's possible that we fail to fetch the update at that specific sequence ID, for example if the server has a stale version of the identity log. I looked at the rest of the code and I don't think we currently attempt to detect if this happens, which is fine given the current server implementation, but we might need to address this eventually.

Heads up @neekolas

Reply:
That's a good call out. We'll have to come up with some way of handling those errors and maybe retrying when we don't have full consistency on messages.